### PR TITLE
Removes attestation pool pruning from UnaggregatedAttestations getter

### PR DESCRIPTION
**Pull request by farazdagi**
**What type of PR is this?**

Other/Refactoring

**What does this PR do? Why is it needed?**
- Currently, we remove seen attestations in `UnaggregatedAttestations()` getter:
https://github.com/prysmaticlabs/prysm/blob/c8e93f87893d41ce7335de0c1ae2c786283d516a/beacon-chain/operations/attestations/kv/unaggregated.go#L61-L65
- Conventionally, getters are to obtain data, not modify it -- otherwise it is a bit unintuitive: you request list of unaggregated attestations, and list gets trimmed (seen attestations are removed). If those seen attestations are unnecessary, and to be removed, then there should be a method specifically for that purpose. This PR introduces such a method.
- So, this PR:
  - [x] removes state amending code from `UnaggregatedAttestations()` getter (seen attestations are still **NOT** returned)
  - [x] adds new `DeleteSeenUnaggregatedAttestations()` method
  - [x] invokes newly defined method in `prune_expired.go`, so that seen attestations are removed, even before the check whether a given unaggregated attestation is expired or not is done.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**
- cc @terencechain

### CODE REVIEW

File: beacon-chain/operations/attestations/prune_expired.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ func (s *Service) pruneExpiredAtts() {
 		}
 	}
 
+	if _, err := s.pool.DeleteSeenUnaggregatedAttestations(); err == nil {
```
Comment (terencechain) [inline beacon-chain/operations/attestations/prune_expired.go:38]: `UnaggregatedAttestations()` is also used in other places. We should be applying `DeleteSeenUnaggregatedAttestations()` to those places as well to rule out any side effects

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/operations/attestations/prune_expired.go (hunk 1 above)
Comment (farazdagi) [reply beacon-chain/operations/attestations/prune_expired.go:38]: Well, when it comes to returned list of attestations - then the very same lists will be returned (the only thing that is changed is side-effect of removing seen attestation is not applied, but seen attestations are not returned).
The `DeleteSeenUnaggregatedAttestations()` is called on a regular basis in `prune_expired.go`, which should be enough?

Comment (terencechain) [reply beacon-chain/operations/attestations/prune_expired.go:38]: Say the following timeline:

1. prune period
2. unaggregated_att_1 gets inserted in the pool
3. duplicated unaggregated_att_1 gets inserted in the pool
4. proposer retrieves unaggregated atts in the pool. (Wouldn't duplicated be used here?)
5. prune period

Comment (farazdagi) [reply beacon-chain/operations/attestations/prune_expired.go:38]: `UnaggregatedAttestations()` will not return duplicates -- they are handled just as before, I just do not delete them at the end. Please, see:
https://github.com/prysmaticlabs/prysm/pull/7204/files#diff-9d7755c9f62602ef4e1eba110d6873e9R60-R61

Or am I missing sth?

Comment (farazdagi) [reply beacon-chain/operations/attestations/prune_expired.go:38]: image

Comment (terencechain) [reply beacon-chain/operations/attestations/prune_expired.go:38]: Ah i see. Agree this is fine then. The only down side is 1 day someone could change prune period to something longer without understanding the implications, but that's out of the scope of this concern. Approving!

**Review by terencechain — APPROVED** [review]

File: beacon-chain/operations/attestations/prune_expired.go (hunk 1 above)
Comment (farazdagi) [reply beacon-chain/operations/attestations/prune_expired.go:38]: Thank you!

**Pull request merged**